### PR TITLE
Spell out secure secret storage and sharing

### PR DESCRIPTION
**Pull request by davidegirardi**
### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview of the change: https://pr1875--matrix-spec-previews.netlify.app/client-server-api/#secrets

Preview: https://pr1875--matrix-spec-previews.netlify.app

### CODE REVIEW

File: content/client-server-api/modules/secrets.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ secret when storing, fetching, requesting, or sharing the secret.
 Secrets are plain strings; structured data can be stored by encoding it
 as a string.
 
+The mechanism described in this section is known as secure secret storage and
+sharing, SSSS, or 4S.
```
Comment (richvdh) [inline content/client-server-api/modules/secrets.md:19]:
```suggestion
The mechanism described in this section is known as "secure secret storage and
sharing", "SSSS", or "4S".
```

File: changelogs/client_server/newsfragments/1875.clarification [hunk 2]
```diff
@@ -0,0 +1 @@
+Include the text secure secret storage and sharing, SSSS, and 4S in the Secrets section to make it easier to find.
```
Comment (richvdh) [inline changelogs/client_server/newsfragments/1875.clarification:1]: This is a bit overly detailed IMHO

```suggestion
Minor clarifications to the "Secrets" section.
```

Comment (turt2live) [reply changelogs/client_server/newsfragments/1875.clarification:1]: we should be describing what the clarifications are.

```suggestion
Document the acronyms and alternate names for the "Secrets" section.
```

**Review by richvdh — APPROVED** [review]
THanks!

**Pull request merged**